columnar_derive/src/lib.rs

```diff
@@ -365,11 +383,15 @@ fn derive_struct(name: &syn::Ident, generics: &syn::Generics, data_struct: syn::
 }
 
 // TODO: Do we need to use the generics?
-fn derive_unit_struct(name: &syn::Ident, _generics: &syn::Generics, vis: syn::Visibility) -> proc_macro::TokenStream {
+fn derive_unit_struct(name: &syn::Ident, _generics: &syn::Generics, vis: syn::Visibility, attr: Option<proc_macro2::TokenStream>) -> proc_macro::TokenStream {
     
     let c_name = format!("{}Container", name);
     let c_ident = syn::Ident::new(&c_name, name.span());
 
+    if attr.is_some() {
+        panic!("Unit structs do not support attributes");
+    }
```

Comment:
Is this a TODO from your point of view, or is there a reason to prevent this?

Reply:
The reason for this is that we don't create a reference type for unit structs, so we can't put the annotation anywhere.

Reply:
This totally makes sense in hindsight! Thanks.